test/common/filesystem/watcher_impl_test.cc

```diff
@@ -13,36 +15,41 @@ TEST(WatcherImplTest, All) {
 
-  unlink("/tmp/envoy_test/watcher_target");
-  unlink("/tmp/envoy_test/watcher_link");
-  unlink("/tmp/envoy_test/watcher_new_target");
-  unlink("/tmp/envoy_test/watcher_new_link");
+  unlink(TestEnvironment::temporaryPath("envoy_test/watcher_target").c_str());
+  unlink(TestEnvironment::temporaryPath("envoy_test/watcher_link").c_str());
+  unlink(TestEnvironment::temporaryPath("envoy_test/watcher_new_target").c_str());
+  unlink(TestEnvironment::temporaryPath("envoy_test/watcher_new_link").c_str());
 
-  mkdir("/tmp/envoy_test", S_IRWXU);
-  { std::ofstream file("/tmp/envoy_test/watcher_target"); }
-  int rc = symlink("/tmp/envoy_test/watcher_target", "/tmp/envoy_test/watcher_link");
+  mkdir(TestEnvironment::temporaryPath("envoy_test").c_str(), S_IRWXU);
+  { std::ofstream file(TestEnvironment::temporaryPath("envoy_test/watcher_target")); }
+  int rc = symlink(TestEnvironment::temporaryPath("envoy_test/watcher_target").c_str(),
+                   TestEnvironment::temporaryPath("envoy_test/watcher_link").c_str());
   RELEASE_ASSERT(0 == rc);
   UNREFERENCED_PARAMETER(rc);
 
-  { std::ofstream file("/tmp/envoy_test/watcher_new_target"); }
-  rc = symlink("/tmp/envoy_test/watcher_new_target", "/tmp/envoy_test/watcher_new_link");
+  { std::ofstream file(TestEnvironment::temporaryPath("envoy_test/watcher_new_target")); }
+  rc = symlink(TestEnvironment::temporaryPath("envoy_test/watcher_new_target").c_str(),
+               TestEnvironment::temporaryPath("envoy_test/watcher_new_link").c_str());
   RELEASE_ASSERT(0 == rc);
 
   WatchCallback callback;
   EXPECT_CALL(callback, called(Watcher::Events::MovedTo)).Times(2);
-  watcher->addWatch("/tmp/envoy_test/watcher_link", Watcher::Events::MovedTo,
-                    [&](uint32_t events) -> void {
+  watcher->addWatch(TestEnvironment::temporaryPath("envoy_test/watcher_link"),
+                    Watcher::Events::MovedTo, [&](uint32_t events) -> void {
                       callback.called(events);
                       dispatcher.exit();
                     });
 
-  rename("/tmp/envoy_test/watcher_new_link", "/tmp/envoy_test/watcher_link");
+  rename(TestEnvironment::temporaryPath("envoy_test/watcher_new_link").c_str(),
+         TestEnvironment::temporaryPath("envoy_test/watcher_link").c_str());
   dispatcher.run(Event::Dispatcher::RunType::Block);
 
-  rc = symlink("/tmp/envoy_test/watcher_new_target", "/tmp/envoy_test/watcher_new_link");
+  rc = symlink(TestEnvironment::temporaryPath("envoy_test/watcher_new_target").c_str(),
+               TestEnvironment::temporaryPath("envoy_test/watcher_new_link").c_str());
   RELEASE_ASSERT(0 == rc);
-  rename("/tmp/envoy_test/watcher_new_link", "/tmp/envoy_test/watcher_link");
+  rename(TestEnvironment::temporaryPath("envoy_test/watcher_new_link").c_str(),
+         TestEnvironment::temporaryPath("envoy_test/watcher_link").c_str());
   dispatcher.run(Event::Dispatcher::RunType::Block);
 }
 
-} // Filesystem
+} // namespace Filesystem
```

Comment:
Not sure how this crept in. The style guide says it should be prefixed but I don't care either. For anonymous namespaces we already do "// namespace" from a quick grep, which is useful as it indicates that there isn't a forgotten label there.